### PR TITLE
stdin works on Windows, too, without regressing issue #5

### LINKED ISSUE #12: Rexi stdin handling does not work on Windows
See https://www.reddit.com/r/Python/comments/1ahw7lj/rexi_a_terminal_ui_for_regex_testing_built_in/kpqx7v4/?context=3

I'm planning on submitting a pull request for this, but wanted to create an issue for it first.

rexi calling `select.select()` on Windows results in this error: "`OSError: [WinError 10038] An operation was attempted on something that is not a socket`"

This error is caused by this commit: https://github.com/royreznik/rexi/commit/e301ba3082b9157da3ee3fc2791867f3293697d4

I believe that this fix is the wrong approach to checking for input. The current behavior is to poll for data in stdin and then raise an error if there's no data.

A better approach would be to check if stdin is a tty. If it is a tty then you can quit with an error that no input is provided, but if it is not a tty (eg `echo "string to match" | rexi` or `rexi <README.me`) then the correct behaviour is to sit and wait for STDIN to terminate. This approach is both similar to how many other tools (like grep) work, and would work on Linux or on Windows.

Other than that, closing the pipe stdin and opening a /dev/tty stdin should work on Windows, but it will require using winpty.PTY instead of /dev/tty.

### PR DESCRIPTION
**Pull request by jshipley**
This fixes #12 

This makes rexi work on Windows again, and is also a good fix for issue #5 where rexi was waiting for input forever when no input was given.

I have run the unit tests and other checks on both Windows and Linux, and everything seems to be working as expected.

I'm not going to guarantee that it will work on any version of Windows older than Windows 10, but given that Windows 10 is EOL in about a year and a half that shouldn't matter.

### CODE REVIEW

**Comment by royreznik** [comment]
Im currently on a vacation for the next few days, so i will do a CR later in the week :)

**Comment by royreznik** [comment]
Seems good to me, Can you fix the coverage to 100%?
Also Can you add a `Windows` to the `Actions` Metrix? if you aren't familiar with it I will add it in another PR

**Comment by jshipley** [comment]
I added windows-latest to the github actions matrix. I'm not really familiar with github actions, so please let me know if more is needed.

**Comment by royreznik** [comment]
It seems to work, but there is a problem in the make file with windows, can you fix it too?
Also please add test that will make it still 100% coverage.
You can tedt the coverage using the 'make test' command and change the coverage parameter to html

**Comment by jshipley** [comment]
I pushed in another change.

For the matrix test, I think that removing "-rf" from the "rm -rf" command should be enough to get it to work. The output from the build makes it look like powershell is running the Windows commands. In powershell, "rm" works, but it runs the "Remove-Item" cmdlet that has different flags than the Linux rm command.

It looks like poetry.lock is a file, not a directory, so the "-rf" was unnecessary. If for some reason this doesn't work, it'll probably need a condition to make it run a different rm command on Windows and Linux.

I ended up writing a simple test case for the new isatty wrapper function, because no matter what I try I can't get a mock for sys.stdin.isatty to work inside the rexi_cli typer function. It might be because of the interaction with typer, or it might be the reassignment of sys.stdin that's happening.

Long story short, test coverage should be 100% now, and the matrix tests should work.

**Comment by jshipley** [comment]
Hmm. It looks like it's failing now because make isn't available on the Windows runner.

I'll add a step to install make, but I don't know of any way to test running a GitHub action without actually checking in the changes and letting GitHub run it.

**Comment by royreznik** [comment]
I can try to add it in another PR, just remove the `action` part and i will play with it later :)

**Comment by royreznik** [comment]
For now lets remove the github action for windows.
It takes a lot of resources and i don't want to use all my `action` time for windows checking :D

**Review by royreznik — APPROVED** [review]
Looks good
